### PR TITLE
start of tagGroup and toc unit tests

### PR DESCRIPTION
**Pull request by lrobledo**
Start of some unit tests for the oas3 converter. 
These are pretty basic—planning to add more soon. I just wanted to get it started. Included a petstore oas 3.0 and data objects for testing.

Also installed nyc for code coverage. Can undo if you prefer!

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Thanks for this, it's in the queue!

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "markdown-it": "^10.0.0",
     "markdown-it-emoji": "^1.4.0",
     "node-fetch": "^2.0.0",
+    "nyc": "^15.0.0",
```
Comment (MikeRalphson) [inline package.json]: `nyc` should be a `devDependency`.

File: lib/openapi3.js [hunk 2]
```diff
@@ -129,7 +129,6 @@ function fakeProdCons(data) {
 }
 
 function getParameters(data) {
-
```
Comment (MikeRalphson) [inline lib/openapi3.js]: Please do not remove blank lines unless they are doubled-up.

**Comment by lrobledo** [comment]
@MikeRalphson fixed!

File: test/openapi3.test.js [hunk 3]
```diff
@@ -75,6 +78,38 @@ describe('openapi3 tests', () => {
...
+
+    describe('getTagGroup', () => {
+        let test = null;
+        it('should handle no tag groups', () => {
+            assert.doesNotThrow(() => openapi3.getTagGroup('fake tag', null));
+        });
+        it('should handle wrong tags ', () => {
+            assert.doesNotThrow(() => openapi3.getTagGroup('fake tag', exampleEnv.tagGroups));
+            let result = openapi3.getTagGroup('fake tag', exampleEnv.tagGroups);
+            assert(result.name === 'fake tag');
```
Comment (MikeRalphson) [inline test/openapi3.test.js:91]: So we get better error messages, I think these `assert` calls should use one of the `*Equals` methods e.g. [assert.deepStrictEqual](https://nodejs.org/api/assert.html#assert_assert_deepstrictequal_actual_expected_message)

Comment (lrobledo) [reply test/openapi3.test.js:91]: No worries, just seeing this now. Will fix this asap

**Comment by lrobledo** [comment]
@MikeRalphson updated those assert functions!

**Comment by lrobledo** [comment]
@MikeRalphson It still says that changes are requested, but I'm pretty sure I fixed the issue. Can you take a look when you get a chance?

**Comment by MikeRalphson** [comment]
Many thanks again, looking forward to future test improvements!

**Pull request merged**